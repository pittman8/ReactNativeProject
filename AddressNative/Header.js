import React, {Component} from 'react';
import {Link} from 'react-router-native';
import {Text, View} from 'react-native';
import styles from "./elf-styles";

class Header extends Component {

    render() {
        return (
            <View style={styles.nav}>
                <Link
                    to="/"
                    underlayColor='#f0f4f7'
                    style={styles.navItem}>
                    <Text>Address</Text>
                </Link>
                <Link
                    to="/get-file"
                    underlayColor='#f0f4f7'
                    style={styles.navItem}>
                    <Text>Get File</Text>
                </Link>
            </View>
        );
    }
}

export default Header;