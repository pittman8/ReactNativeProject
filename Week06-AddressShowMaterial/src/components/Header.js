import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import '../css/menu.css';
import '../css/App.css';
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';

class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false
        };
    }

    handleToggle = () => this.setState({ open: !this.state.open });

    render() {
        return (
            <div className="Header">
                <AppBar
                    title="Address Maven"
                    iconClassNameRight="muidocs-icon-navigation-expand-more"
                    onLeftIconButtonClick={this.handleToggle}
                />
                <Drawer
                    docked={false}
                    width={200}
                    open={this.state.open}
                    onRequestChange={this.handleToggle}
                >
                    <AppBar title="Address Maven" />

                    <MenuItem
                        primaryText="Address"
                        containerElement={<Link to="/" />}
                        onClick={this.handleToggle}
                    />

                    <MenuItem
                        primaryText="Get File"
                        containerElement={<Link to="/get-file" />}
                        onClick={this.handleToggle}
                    />
                </Drawer>
            </div>
        );
    }
}
export default Header;
