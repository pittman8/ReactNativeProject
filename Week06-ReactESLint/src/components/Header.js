import React, { Component } from "react";
import logo from "../logo.svg";
import { Link } from "react-router-dom";
import "../css/menu.css";

class Header extends Component {
  render() {
    return (
      <div className="Header">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Address Maven</h1>
        </header>
        <ul>
          <li>
            <Link to="/">Address</Link>
          </li>
          <li>
            <Link to="/get-file">Get File</Link>
          </li>
        </ul>
      </div>
    );
  }
}
export default Header;
