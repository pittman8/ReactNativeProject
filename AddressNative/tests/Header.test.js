import React from 'react';
import Header from '../Header';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Text } from 'react-native';

configure({ adapter: new Adapter() });

describe('Header tests', function() {
    it('renders without crashing', () => {
        const rendered = renderer.create(<Header />).toJSON();
        expect(rendered).toBeTruthy();
    });

    it('renders default output with react (mount)', () => {
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('finds a text element', () => {
        const wrapper = shallow(<Header />);
        const link = <Text>Address</Text>;
        expect(wrapper.dive().contains(link)).toBe(true);
    });
});
