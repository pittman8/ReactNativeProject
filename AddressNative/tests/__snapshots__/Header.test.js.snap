// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders default output with react (mount) 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={null}
    >
      Address
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={null}
    >
      Get File
    </Text>
  </View>
</View>
`;
